### PR TITLE
feat(cmd/helm): remove need for helm init command

### PR DESCRIPTION
**Pull request by adamreese**
* allow repository config via cli
* make `helm repo add` create repo config file if it does not exist
* squash a ton of bugs
* remove init command

### CODE REVIEW

File: pkg/repo/chartrepo.go [hunk 1]
```diff
@@ -114,31 +108,37 @@ func (r *ChartRepository) Load() error {
...
+		return "", err
 	}
 
 	if _, err := loadIndex(index); err != nil {
-		return err
+		return "", err
 	}
 
-	return ioutil.WriteFile(helmpath.CacheIndex(r.Config.Name), index, 0644)
+	fname := filepath.Join(r.CachePath, r.Config.Name+"-index.yaml")
```
Comment (thomastaylor312) [inline pkg/repo/chartrepo.go]: Should this be some functionality inside of `helmpath` or is this just a one off?

**Review by thomastaylor312 — CHANGES_REQUESTED** [review]
Just a few questions and some non-blocking nits

File: cmd/helm/repo_add.go [hunk 2]
```diff
@@ -68,43 +75,44 @@ func newRepoAddCmd(out io.Writer) *cobra.Command {
...
 	}
-	fmt.Fprintf(out, "%q has been added to your repositories\n", o.name)
-	return nil
-}
 
-func addRepository(name, url, username, password string, certFile, keyFile, caFile string, noUpdate bool) error {
-	f, err := repo.LoadFile(helmpath.RepositoryFile())
-	if err != nil {
+	var f repo.File
+	if err := yaml.Unmarshal(b, &f); err != nil {
```
[the remark this answers is not in the document]
Comment (adamreese) [reply cmd/helm/repo_add.go]: Because we want to handle the case of no file existing.

**Review by thomastaylor312 — APPROVED** [review]
LGTM

**Pull request merged**